### PR TITLE
[GeoMechanicsApplication] Simplified fluid flux for pw element

### PR DESCRIPTION
**Pull request by rfaasse**
**📝 Description**
We found that to calculate the fluid flux variable, the permeability update factor was calculated using strains, regardless of Pw vs UPw. Since strains are meaningless for Pw elements, this needed to be rectified. Therefore, a function is created to calculate fluid fluxes, which takes as an input a list of permeability update factors for the integration points (which are now always 1 for the Pw elements and calculated via the strain for UPw elements).

### CODE REVIEW

File: applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp [hunk 1]
```diff
@@ -1112,6 +1091,48 @@ void UPwSmallStrainElement<TDim, TNumNodes>::InitializeBiotCoefficients(ElementV
...
+        GeoElementUtilities::InterpolateVariableWithComponents<TDim, TNumNodes>(
+            Variables.BodyAcceleration, Variables.NContainer, Variables.VolumeAcceleration, GPoint);
+        Variables.FluidPressure = CalculateFluidPressure(Variables);
+        RetentionParameters.SetFluidPressure(Variables.FluidPressure);
+
+        Variables.RelativePermeability =
+            mRetentionLawVector[GPoint]->CalculateRelativePermeability(RetentionParameters);
+
+        noalias(GradPressureTerm) = prod(trans(Variables.GradNpT), Variables.PressureVector);
+
```
Comment (WPK4FEM) [inline applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp]: lines 1124 and 1126 are constructing the contents of the same variable. This interspacing contradicts that.

**Review by WPK4FEM — DISMISSED** [review]
Hi Richard,
The intent is clear to me.
Only some nagging about layout.
Wijtze Pieter

File: applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp [hunk 2]
```diff
@@ -1112,6 +1091,48 @@ void UPwSmallStrainElement<TDim, TNumNodes>::InitializeBiotCoefficients(ElementV
     KRATOS_CATCH("")
 }
 
+template <unsigned int TDim, unsigned int TNumNodes>
+std::vector<array_1d<double, TDim>> UPwSmallStrainElement<TDim, TNumNodes>::CalculateFluidFluxes(
+    const std::vector<double>& permeability_update_factors, const ProcessInfo& rCurrentProcessInfo)
```
Comment (avdg81) [inline applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp]: To be in line with the Kratos Style Guide, we should rename the first parameter:
```suggestion
    const std::vector<double>& rPermeabilityUpdateFactors, const ProcessInfo& rCurrentProcessInfo)
```

**Review by avdg81 — COMMENTED** [review]
Nice improvement of the code. I have only one very minor comment. Let's try to get this in quickly, so you can proceed with the PR that started all this (extracting lists of $B$ matrices).

File: applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp (hunk 2 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp]: Done

File: applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp (hunk 1 above)
Comment (rfaasse) [reply applications/GeoMechanicsApplication/custom_elements/U_Pw_small_strain_element.cpp]: Removed blank line

**Review by WPK4FEM — APPROVED** [review]
Resize kijkt zelf wel naar de oorspronkelijke grootte, daar hoeft geen test voor en hoeft ook niet herhaald te worden met dezelfde grootte met een andere naam.
Dat kan best in een volgende ronde, want we zijn hier toch nog uitgebreid bezig.

**Pull request merged**